agent/util-scripts/pbench-tool-meister-stop

```diff
@@ -103,20 +181,38 @@ def main(_prog, cli_params):
         full_hostname = os.environ["_pbench_full_hostname"]
         benchmark_run_dir = Path(os.environ["benchmark_run_dir"]).resolve(strict=True)
     except Exception:
-        logger.exception("failed to fetch parameters from the environment")
+        logger.exception("failed to fetch required parameters from the environment")
         return 1
 
-    try:
-        redis_server_pid_file = (
-            benchmark_run_dir / "tm" / f"redis_{redis_port:d}.pid"
-        ).resolve(strict=True)
-    except FileNotFoundError:
-        # No Redis server, nothing to do.
-        return 0
+    if cli_params.redis_server is None:
+        # No Redis server was given, so look locally to see if we can find it.
+        # If no Redis server locally, we're done.
+        try:
+            redis_server_pid_file = (
+                benchmark_run_dir / "tm" / f"redis_{def_redis_port:d}.pid"
+            ).resolve(strict=True)
+        except FileNotFoundError:
+            # No Redis server, nothing to do.
+            return 0
+        else:
+            redis_host = "localhost"
+            redis_port = def_redis_port
+    else:
+        parts = cli_params.redis_server.split(":", 1)
+        if len(parts) != 2:
+            logger.error("Bad Redis server specified, '%s'", cli_params.redis_server)
+            return 1
+        try:
+            redis_port = int(parts[1])
+        except Exception:
+            logger.error("Bad Redis port specified, '%s'", cli_params.redis_server)
+            return 1
+        else:
+            redis_host = parts[0]
```

Comment:
Do we want to support ":55555" as a shorthand for "localhost:55555"?

Reply:
Y'know, this code should be refactored so that we get the same support for `-start`, `-stop`, and `-client`....

Reply:
> Do we want to support ":55555" as a shorthand for "localhost:55555"?

Tracked in #2170.

Reply:
> Y'know, this code should be refactored so that we get the same support for `-start`, `-stop`, and `-client`....

Tracked in https://github.com/distributed-system-analysis/pbench/issues/2176.